### DISCUSSION TITLE
trying to get a new refresh_token

### DISCUSSION THREAD

**Discussion by armup31** [nwunderly/starlette-discord#26] [Q&A] [answered]
```py
user = request.session.get("discord_user")
if not user:
  return {"status": "error", "message": "User not logged in"}
sesh = client.session_from_token(user["token"])
new_token = await sesh.refresh_token(token_url="https://discord.com/api/v10/oauth2/token/")
```

Hello guys im kinda new into this but I'm trying to get a new refresh_token, but when i do so, i get an error 
`raise MissingTokenError(description="Missing access token parameter.")
oauthlib.oauth2.rfc6749.errors.MissingTokenError: (missing_token) Missing access token parameter.`

I tried searching on google, but did not find anything and there isn't an example for the refresh_token func either. So please help me!

**Comment by BeeMoe5** [comment, marked as answer]
Under the assumption you are using version 0.2.1, there should a method on session `.refresh()` that should be the one that works. `.refresh_token()` is the one we stole from someone's generic implementation on an oauth2 client. Let us know about any further issues!

**Reply by armup31** [reply]
It's the one shown in the documentation so that's what I implemented but thanks!

**Reply by BeeMoe5** [reply]
> It's the one shown in the documentation so that's what I implemented but thanks!

Yeah sorry for the confusion, its auto generated docs from our repo's code so it included `.refresh_token()`. We plan on updating our codebase so the auto generated docs will be more clear and won't mislead anyone lol
